diff --git a/flashrom.c b/flashrom.c
index f9e67b15..9c25b5e4 100644
--- a/flashrom.c
+++ b/flashrom.c
@@ -2364,11 +2364,6 @@ int prepare_flash_access(struct flashctx *const flash,
 		return 1;
 	}
 
-	/* FIXME(b/207787495): replace this with locking in futility. */
-	/* Let powerd know that we're updating firmware so machine stays awake. */
-	if (write_it || erase_it)
-		disable_power_management();
-
 	if (map_flash(flash) != 0)
 		return 1;
 
@@ -2418,11 +2413,6 @@ void finalize_flash_access(struct flashctx *const flash)
 {
 	deregister_chip_restore(flash);
 	unmap_flash(flash);
-
-	/* FIXME(b/207787495): replace this with locking in futility. */
-	if (restore_power_management()) {
-		msg_gerr("Unable to re-enable power management\n");
-	}
 }
 
 static int setup_curcontents(struct flashctx *flashctx, void *curcontents,
