diff --git a/make_helpers/build_macros.mk b/make_helpers/build_macros.mk
index d27408c50..e03a6085b 100644
--- a/make_helpers/build_macros.mk
+++ b/make_helpers/build_macros.mk
@@ -307,7 +307,7 @@ $(eval DEP := $(patsubst %.o,%.d,$(OBJ)))
 
 $(OBJ): $(2) $(filter-out %.d,$(MAKEFILE_LIST)) | lib$(3)_dirs
 	$$(s)echo "  AS      $$<"
-	$$(q)$($(ARCH)-as) -x assembler-with-cpp $$(TF_CFLAGS_$(ARCH)) $$(ASFLAGS) $(MAKE_DEP) -c $$< -o $$@
+	$$(q)$($(ARCH)-cc) -x assembler-with-cpp $$(TF_CFLAGS_$(ARCH)) $$(ASFLAGS) $(MAKE_DEP) -c $$< -o $$@
 
 -include $(DEP)
 
@@ -353,7 +353,7 @@ $(eval BL_ASFLAGS := $($(call uppercase,$(3))_ASFLAGS) $(PLAT_BL_COMMON_ASFLAGS)
 
 $(OBJ): $(2) $(filter-out %.d,$(MAKEFILE_LIST)) | $(3)_dirs
 	$$(s)echo "  AS      $$<"
-	$$(q)$($(ARCH)-as) -x assembler-with-cpp $$(TF_CFLAGS_$(ARCH)) $$(ASFLAGS) $(BL_CPPFLAGS) $(BL_ASFLAGS) $(MAKE_DEP) -c $$< -o $$@
+	$$(q)$($(ARCH)-cc) -x assembler-with-cpp $$(TF_CFLAGS_$(ARCH)) $$(ASFLAGS) $(BL_CPPFLAGS) $(BL_ASFLAGS) $(MAKE_DEP) -c $$< -o $$@
 
 -include $(DEP)
 
@@ -564,11 +564,11 @@ endif
 
 $(DUMP): $(ELF)
 	$$(s)echo "  OD      $$@"
-	$$(q)$($(ARCH)-od) -dx $$< > $$@
+	$$(q)$($(ARCH)-objdump) -dx $$< > $$@
 
 $(BIN): $(ELF)
 	$$(s)echo "  BIN     $$@"
-	$$(q)$($(ARCH)-oc) -O binary $$< $$@
+	$$(q)$($(ARCH)-objcopy) -O binary $$< $$@
 	$(s)echo
 	$(s)echo "Built $$@ successfully"
 	$(s)echo
diff --git a/make_helpers/toolchains/aarch64.mk b/make_helpers/toolchains/aarch64.mk
index 407f06818..dfc4fd32a 100644
--- a/make_helpers/toolchains/aarch64.mk
+++ b/make_helpers/toolchains/aarch64.mk
@@ -11,15 +11,15 @@ aarch64-cc-default := $(or $(CROSS_COMPILE),aarch64-none-elf-)gcc
 aarch64-cc-id-default := gnu-gcc
 
 aarch64-cpp := $(if $(filter-out default,$(origin CPP)),$(CPP))
-aarch64-cpp-default := $(or $(CROSS_COMPILE),aarch64-none-elf-)gcc
+aarch64-cpp-default := $(or $(CROSS_COMPILE),aarch64-none-elf-)g++
 aarch64-cpp-id-default := gnu-gcc
 
 aarch64-as := $(if $(filter-out default,$(origin AS)),$(AS))
-aarch64-as-default := $(or $(CROSS_COMPILE),aarch64-none-elf-)gcc
+aarch64-as-default := $(or $(CROSS_COMPILE),aarch64-none-elf-)as
 aarch64-as-id-default := gnu-gcc
 
 aarch64-ld := $(if $(filter-out default,$(origin LD)),$(LD))
-aarch64-ld-default := $(or $(CROSS_COMPILE),aarch64-none-elf-)gcc
+aarch64-ld-default := $(or $(CROSS_COMPILE),aarch64-none-elf-)ld
 aarch64-ld-id-default := gnu-gcc
 
 aarch64-oc := $(if $(filter-out default,$(origin OC)),$(OC))
@@ -31,7 +31,7 @@ aarch64-od-default := $(or $(CROSS_COMPILE),aarch64-none-elf-)objdump
 aarch64-od-id-default := gnu-objdump
 
 aarch64-ar := $(if $(filter-out default,$(origin AR)),$(AR))
-aarch64-ar-default := $(or $(CROSS_COMPILE),aarch64-none-elf-)gcc-ar
+aarch64-ar-default := $(or $(CROSS_COMPILE),aarch64-none-elf-)ar
 aarch64-ar-id-default := gnu-ar
 
 aarch64-dtc := $(if $(filter-out default,$(origin DTC)),$(DTC))
