diff -Naur a/drivers/char/tpm/tpm_tis_spi_main.c b/drivers/char/tpm/tpm_tis_spi_main.c
--- a/drivers/char/tpm/tpm_tis_spi_main.c	2025-03-12 19:09:36.664664523 -0700
+++ b/drivers/char/tpm/tpm_tis_spi_main.c	2025-03-12 19:10:33.484661242 -0700
@@ -350,7 +350,11 @@
 		.pm = &tpm_tis_pm,
 		.of_match_table = of_match_ptr(of_tis_spi_match),
 		.acpi_match_table = ACPI_PTR(acpi_tis_spi_match),
+#ifdef CONFIG_IMA
+		.probe_type = PROBE_FORCE_SYNCHRONOUS
+#else
 		.probe_type = PROBE_PREFER_ASYNCHRONOUS,
+#endif
 	},
 	.probe = tpm_tis_spi_driver_probe,
 	.remove = tpm_tis_spi_remove,
diff -Naur a/drivers/spi/spi.c b/drivers/spi/spi.c
--- a/drivers/spi/spi.c	2025-03-12 19:09:37.054664199 -0700
+++ b/drivers/spi/spi.c	2025-03-12 19:10:47.975673817 -0700
@@ -4957,4 +4957,4 @@
  * driver registration) _could_ be dynamically linked (modular) ... Costs
  * include needing to have boardinfo data structures be much more public.
  */
-postcore_initcall(spi_init);
+core_initcall(spi_init);
