From 9bb6b834273324c806f04a2dc607daf0bcc092f6 Mon Sep 17 00:00:00 2001
From: Jared Baur <jaredbaur@fastmail.com>
Date: Sat, 18 Nov 2023 09:04:25 -0800
Subject: [PATCH 2/3] Fix build for brya

---
 src/mainboard/google/brya/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mainboard/google/brya/Kconfig b/src/mainboard/google/brya/Kconfig
index 4a38af5f2c..9bbdade7a6 100644
--- a/src/mainboard/google/brya/Kconfig
+++ b/src/mainboard/google/brya/Kconfig
@@ -730,7 +730,7 @@ config VARIANT_DIR
 
 config VBOOT
 	select VBOOT_EARLY_EC_SYNC if !BOARD_GOOGLE_BASEBOARD_NISSA
-	select VBOOT_LID_SWITCH
+	select VBOOT_LID_SWITCH if CHROMEOS
 
 config DIMM_SPD_SIZE
 	default 512
-- 
2.43.1

