diff -Naur a/linux-6.8.9/drivers/char/tpm/tpm_tis_spi_main.c b/linux-6.8.9/drivers/char/tpm/tpm_tis_spi_main.c
--- a/drivers/char/tpm/tpm_tis_spi_main.c	2024-05-02 14:35:35.000000000 +0000
+++ b/drivers/char/tpm/tpm_tis_spi_main.c	2024-05-11 01:32:52.467923228 +0000
@@ -347,7 +347,11 @@
 		.pm = &tpm_tis_pm,
 		.of_match_table = of_match_ptr(of_tis_spi_match),
 		.acpi_match_table = ACPI_PTR(acpi_tis_spi_match),
+#ifdef CONFIG_IMA
+		.probe_type = PROBE_FORCE_SYNCHRONOUS
+#else
 		.probe_type = PROBE_PREFER_ASYNCHRONOUS,
+#endif
 	},
 	.probe = tpm_tis_spi_driver_probe,
 	.remove = tpm_tis_spi_remove,
diff -Naur a/linux-6.8.9/drivers/spi/spi.c b/linux-6.8.9/drivers/spi/spi.c
--- a/drivers/spi/spi.c	2024-05-02 14:35:35.000000000 +0000
+++ b/drivers/spi/spi.c	2024-05-11 01:33:18.267023406 +0000
@@ -4832,4 +4832,4 @@
  * driver registration) _could_ be dynamically linked (modular) ... Costs
  * include needing to have boardinfo data structures be much more public.
  */
-postcore_initcall(spi_init);
+core_initcall(spi_init);
